### PR TITLE
test: update context happy path

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Modified the integration test that now starts with a feature in a disabled state (due to context settings) and then goes to enabled state (due to another context update).

Here's the API settings for the toggle used in this test:
Screenshot 2023-11-17 at 10 29 00

Also added test for variant when the toggle is enabled. Here's the settings for the variant (at the strategy level):
Screenshot 2023-11-17 at 10 29 06

### CODE REVIEW

File: Tests/UnleashProxyClientSwiftTests/UnleashClientIntegrationTest.swift [hunk 1]
```diff
@@ -19,31 +19,43 @@ class UnleashIntegrationTests: XCTestCase {
...
+                XCTAssertTrue(variant.enabled, "Variant should be enabled")
+                XCTAssert(variant.name == "feature-variant")
+                
+                expectation.fulfill()
+            }
         });
 
+        unleashClient.start()
+
+        self.unleashClient.updateContext(context: ["clientId": "disabled"]);
```
Comment (kwasniew) [inline Tests/UnleashProxyClientSwiftTests/UnleashClientIntegrationTest.swift:57]: we want to start in the disabled state that we control with a context field

File: Tests/UnleashProxyClientSwiftTests/UnleashClientIntegrationTest.swift [hunk 2]
```diff
@@ -19,31 +19,43 @@ class UnleashIntegrationTests: XCTestCase {
...
-            
-            expectation.fulfill()
+            if updateCount == 1 {
+                XCTAssertFalse(self.unleashClient.isEnabled(name: self.featureName), "Feature should be disabled")
+
+                let variant = self.unleashClient.getVariant(name: self.featureName)
+                XCTAssertNotNil(variant, "Variant should not be nil")
+                XCTAssertFalse(variant.enabled, "Variant should be disabled")
+
+                self.unleashClient.updateContext(context: ["clientId": "enabled"]);
```
Comment (kwasniew) [inline Tests/UnleashProxyClientSwiftTests/UnleashClientIntegrationTest.swift:44]: after the first update we change the context to trigger the enabled state

File: Tests/UnleashProxyClientSwiftTests/UnleashClientIntegrationTest.swift [hunk 3]
```diff
@@ -19,31 +19,43 @@ class UnleashIntegrationTests: XCTestCase {
...
+            } else if updateCount == 2 {
+                XCTAssertTrue(self.unleashClient.isEnabled(name: self.featureName), "Feature should be enabled")
```
Comment (kwasniew) [inline Tests/UnleashProxyClientSwiftTests/UnleashClientIntegrationTest.swift:46]: finally feature should become enabled after the update context

**Review by gastonfournier — APPROVED** [review]
LG!

**Pull request merged**